cpp/SnowflakeS3Client.cpp

```diff
@@ -107,8 +107,12 @@ SnowflakeS3Client::SnowflakeS3Client(StageInfo *stageInfo,
   clientConfiguration.caFile = caFile;
   clientConfiguration.requestTimeoutMs = 40000;
   clientConfiguration.connectTimeoutMs = 30000;
-  if(transferConfig != nullptr && transferConfig->useS3regionalUrl)
-  {
+
+  // FIPS mode check
+  if (!(m_stageInfo->endPoint.empty())) {
```

Comment:
Don't you also need to check to make sure `endPoint` isn't null before calling `.empty()` ?

Reply:
I don't think so - [Here's](https://github.com/snowflakedb/libsnowflakeclient/blob/efde8ea8df69bc4dbe7d7e14c93d331be9879c28/include/snowflake/PutGetParseResponse.hpp#L62) the definition of `string StageInfo` which `endPoint` is a member of.  `endPoint` itself isn't a pointer.